src/app/(core)/groups/(overview)/page.tsx

```diff
@@ -1,3 +1,171 @@
+'use client';
+import React, { useState, useEffect } from 'react';
+import GroupCard from '@/components/GroupCard';
+import { createClient } from '../../../../utils/supabase/client';
+import Image from 'next/image';
+import Button from '@/components/ui/button';
+import { HiMagnifyingGlass } from "react-icons/hi2";
+import BellIcon from "../../../../../public/assets/icons/BellIcon.svg";
+import MyGroups from '@/components/MyGroups';
+import { FaChevronRight, FaChevronLeft } from 'react-icons/fa';
+import { PiUsersFour } from "react-icons/pi";
+import { CiCalculator1 } from "react-icons/ci";
+import { HiOutlineBookOpen } from "react-icons/hi";
+import { PiGlobeStand } from "react-icons/pi";
+import { PiLaptop } from "react-icons/pi";
+import { LuLanguages } from "react-icons/lu";
+import Science from "../../../../../public/assets/icons/Science.svg";
+import { getLoggedInUser } from '@/utils/actions/user.actions';
+
+const Subjects = [
+  {
+    id: 1,
+    name: "All Groups",
+    icon: PiUsersFour,
+  },
+  {
+    id: 2,
+    name: "Math",
+    icon: CiCalculator1,
+  },
+  {
+    id: 3,
+    name: "Science",
+    icon: Science,
+  },
+  {
+    id: 4,
+    name: "English",
+    icon: HiOutlineBookOpen,
+  },
+  {
+    id: 5,
+    name: "Social Science",
+    icon: PiGlobeStand,
+  },
+  {
+    id: 6,
+    name: "Computing",
+    icon: PiLaptop,
+  },
+  {
+    id: 7,
+    name: "Languages",
+    icon: LuLanguages,
+  },
+];
+
 export default function GroupsPage(): JSX.Element {
-  return <div>GroupsPage</div>
-}
+  const supabase = createClient();
+  const [groups, setGroups] = useState<any[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [selectedSubject, setSelectedSubject] = useState<string>('All Groups');
+  const [admins, setAdmins] = useState<{ [key: string]: { name: string; avatar: string } }>({});
```

Comment:
can you make a type instead of putting that directly?

Reply:
Here is a step-by-step guide that will help you remember all of the essential steps of writing a code review.

**How to communicate?**

- Instead of giving orders, make suggestions. "I think it would be better to use semantic tags" sounds nicer than "Use semantic tags," doesn't it?
- Remember to explain why you request a change - it might be obvious for you but not for someone else.
- Remember to **be polite** and clear in your comments - your job here is to help another person improve in a friendly way. ALWAYS use the magic word "Please..."
- You can include emojis and gifs to make your code review more human-friendly 👽.

Reply:
Done